cyclonedds/tools/cli/data/__init__.py

```diff
@@ -1,19 +1,22 @@
 import time
-from typing import Any
+from typing import Any, Optional
 from cyclonedds import qos, domain, sub, topic
 
 from ..utils import LiveData
 
 
 def subscribe(
     live: LiveData,
-    domain_id: int,
+    domain_id: Optional[int],
     topic_name: str,
     datatype: Any,
     topicqos: qos.Qos,
     readerqos: qos.Qos,
 ):
-    dp = domain.DomainParticipant(domain_id)
+    if domain_id is None:

```

Comment:
Pardon my ignorance, but why can't you pass `domain_id` directly to `domain.DomainParticipant`?

Reply:
The DomainParticipant constructor was originally written with the expectation that users would either pass an int for the domain_id parameter or omit the parameter which would then cause the domain_id to default to a value of 0. My intention was to change this so it defaulted to DDS_DOMAIN_DEFAULT instead but looking back at the code I've made a mistake with the type hinting as the Optional hint allows None to be passed which wasn't the intention. I've corrected the type hinting for the constructor back to its original form which conveys this intent.

In the CLI tooling, however, I did want the functions to accept a domain_id of None which was intended to indicate that the domain Id argument had not been specified at the command line. Passing this None value through to the DomainParticipant constructor would actually result in the domain_id parameter being set to None instead of the default value which is not desired.

Also on reflection, I think these if/else DomainParticipant creation statements can be made more concise through the use of ternary statements such as:

dp = domain.DomainParticipant(domain_id) if domain_id else domain.DomainParticiant()

If you prefer this syntax I can update the PR.

Reply:
I should also add that Click (the library used for handling the CLI arguments) passes an argument of None by default to the handling function when users imit the option.

Reply:
@eboasson have you had a chance to take another look at this?